### PR TITLE
use hadolint for linting Dockerfiles

### PR DESCRIPTION
**Pull request by fbartels**
- add hadolint to travisfile

fixes https://github.com/dani-garcia/bitwarden_rs/issues/522

### CODE REVIEW

**Comment by dani-garcia** [comment]
The command is making the build fail because it returns a non zero error code, we could just do something like `command || true` but then the warnings would be hidden unless someone checks travis log. Maybe there is a way to tell travis to just warn on error for a specific step?

**Comment by fbartels** [comment]
Yes, i would even say the failure is exactly the idea. I will take care to resolve all errors and will remove the "WIP" once this is ready to be squashed and merged.

**Comment by fbartels** [comment]
@dani-garcia with the last commit the build should now succeed. There is quite a lot of duplication in the Dockerfiles, I am wondering if this could be simplified a bit (for the different architectures for example) when using build arguments.

File: docker/aarch64/mysql/Dockerfile [hunk 1]
```diff
@@ -2,32 +2,35 @@
...
 
-RUN apk add --update-cache --upgrade \
+RUN apk add --no-cache --upgrade \
     curl \
     tar
 
 RUN mkdir /web-vault
 WORKDIR /web-vault
 
+SHELL ["/bin/ash", "-eo", "pipefail", "-c"]
```
Comment (dani-garcia) [inline docker/aarch64/mysql/Dockerfile]: What does this line do exactly?

**Comment by dani-garcia** [comment]
Yeah it would be great if the setup could be simplified a little bit, but I don't know enough about docker or what docker hub supports to do it myself.

File: docker/aarch64/mysql/Dockerfile (hunk 1 above)
Comment (fbartels) [reply docker/aarch64/mysql/Dockerfile]: The following line uses a pipe (curl | tar). This line makes the docker build fail if the pipe fails.

**Comment by mprasil** [comment]
@fbartels we actually attempted to use the build arguments [and it didn't work](https://github.com/dani-garcia/bitwarden_rs/issues/512#issuecomment-504917374). I actually think we could benefit from multiple Dockerfiles and limit the dependencies being installed.

**Comment by fbartels** [comment]
Yes, I've also done a bit of research meanwhile and when just relying on the Docker Hub and not travis in combination with some other scripting (make or a shell script) this does not really seem possible. I was not even referring to sqlite backend vs. mysql backen, but was just thinking about the builds for the different architectures.

But in general I would say that the pr is complete for the moment.

**Comment by dani-garcia** [comment]
Great, thanks for your work, @fbartels!

**Pull request merged**